bigtable/client/internal/endian.h

```diff
@@ -0,0 +1,139 @@
+ * @endcode
+ *
+ * Convert from string of BigEndian bytes to Numeric Value.
+ * @code
+ * std::int64_t cell_value = new_cell.value_as<bigtable::bigendian64_t>().get();
+ * @endcode
+ *
+ * How to use to support new data types:
+ * Recommended way is to define your own Strong Type
+ * (see: bigtable/internal/strong_type.h) to use with this struct.
+ *
+ *  // Following code shows how to support new type for Encode and Decode
+ *  // functions. Code shows support fot bigtable::bigendian64_t strong type
+ *
+ *  // Encode function : Convert BigEndian 64 bit hex bytes inside
+ *  // strong type bigendian64_t into string of bytes.
+ * @code
+ * template<>
+ * std::string Encoder<bigtable::bigendian64_t>::Encode
+ *                               (bigtable::bigendian64_t const& value) {}
+ *  // Decode function : Convert BigEndian 64 bit string of bytes into
+ *  // strong type bigendian64_t
+ * template<>
+ * bigtable::bigendian64_t Encoder<bigtable::bigendian64_t>::Decode
+ *                               (std::string const& value) {}
+ * @endcode
+ *
+ * Implement above two functions for conversion and use it as mentioned
+ * above how to use example.
+ */
+template <typename T>
+struct Encoder {
+  /**
+   * Convert BigEndian numeric value into a string of bytes and return it.
+   *
+   * Google Cloud Bigtable stores arbitrary blobs in each cell. These
+   * blobs are stored in a cell as a string of bytes. Values need to be
+   * converted to/from these cell blobs bytes to be used in application.
+   * This function is used to convert BigEndian 64 bit numeric value into
+   * string of bytes, so that it could be stored as a cell blob.
+   * For this conversion we consider the char size is 8 bits.
+   * So If machine architecture does not define char as 8 bit then we
+   * raise the assert.
+   *
+   * @param value const reference to a number need to be converted into
+   *        string of bytes.
+   */
+  static std::string Encode(T const& value);
+
+  /**
+   * Convert BigEndian string of bytes into BigEndian numeric value and
+   * return it.
+   *
+   * Google Cloud Bigtable stores arbitrary blobs in each cell. These
+   * blobs are stored in a cell as a string of bytes. Values need to be
+   * converted to/from these cell blobs bytes to be used in application.
+   * This function is used to convert string of bytes into BigEndian 64
+   * bit numeric value. so that it could be used in application.
+   * For this conversion we consider the char size is 8 bits.
+   * So If machine architecture does not define char as 8 bit then we
+   * raise the std::range_error.
+   *
+   * @param value reference to a string of bytes need to be converted into
+   *        BigEndian numeric value.
+   */
+  static T Decode(std::string const& value);
+};
+
+bigtable::bigendian64_t byteswap64(bigtable::bigendian64_t value);
+
+}  // namespace internal
+
+std::string as_bigendian64(bigtable::bigendian64_t value);
```

Comment:
Can we use `Decoder<bigendian64_t>::Encode` and not define this function at all?

Reply:
I think we should keep this function as it is very handy instead of using bigtable::internal::Encoder<bigtable::bigendian64_t>::Encode.
I think we should also have a similar function for bigtable::internal::Encoder<T>::Decode, 
but I am not getting convenient name for it.
what do you think?